### PR TITLE
AutoYaST documentation update

### PR DESCRIPTION
**Pull request by imobachgs**
Update AutoYaST and storage-ng documentation. See https://trello.com/c/SjtD2G62/1403-3-step-1-clarify-update-autoyast-documentation-regarding-storage.

### CODE REVIEW

File: doc/autoyast.md [hunk 1]
```diff
@@ -170,29 +203,42 @@ Note that there are no separate classes for {Y2Storage::Planned::Disk} and
...
+
 ## Issues reporting
 
 It is possible that, given a profile, AutoYaST finds issues when trying to
-figure out a the partitioning layout. Some of those issues might be serious
-enough to stop the installation; in other cases, just displaying a warning could
-be the way to go.
+figure out the partitioning layout. Some of those issues might be serious
+enough to stop the installation; in other cases, just displaying a warning is
+the way to go.
```
Comment (jreidinger) [inline doc/autoyast.md]: maybe I would simplify it to `Some of those issues might be serious
￼enough to stop the installation; in other cases, it just displays a warning.`

File: doc/autoyast.md [hunk 2]
```diff
@@ -170,29 +203,42 @@ Note that there are no separate classes for {Y2Storage::Planned::Disk} and
...
+[4]: https://github.com/yast/yast-autoinstallation/blob/942413b8b54171ec3a79884c9be2138d11ba6803/src/modules/AutoinstStorage.rb "AutoinstStorage module"
+[5]: https://github.com/yast/yast-autoinstallation/blob/942413b8b54171ec3a79884c9be2138d11ba6803/src/modules/AutoinstStorage.rb#L75 "AutoinstStorage#import_general_settings"
+[6]: https://github.com/yast/yast-autoinstallation/blob/942413b8b54171ec3a79884c9be2138d11ba6803/src/modules/AutoinstStorage.rb#L316 "AutoinstStorage#preprocessed_settings"
+[7]: https://github.com/yast/yast-autoinstallation/blob/942413b8b54171ec3a79884c9be2138d11ba6803/src/modules/AutoinstStorage.rb#L251 "AutoinstStorage#build_proposal"
+[8]: https://github.com/yast/yast-autoinstallation/blob/942413b8b54171ec3a79884c9be2138d11ba6803/src/modules/AutoinstStorage.rb#L272 "AutoinstStorage#valid_proposal?"
+[9]: https://github.com/yast/yast-autoinstallation/blob/942413b8b54171ec3a79884c9be2138d11ba6803/src/modules/AutoinstStorage.rb#L296 "Present issues to the user"
+[10]: https://github.com/yast/yast-autoinstallation/blob/942413b8b54171ec3a79884c9be2138d11ba6803/src/lib/autoinstall/storage_proposal.rb "StorageProposal"
```
Comment (jreidinger) [inline doc/autoyast.md]: I am not sure if it is better to link to fixed commit that can be really outdated or link to master, when lines may not fit.

File: doc/autoyast.md [hunk 3]
```diff
@@ -2,92 +2,124 @@
...
-   are partitions have to be set.
-3. Expert partitioning. The user specifies how the layout should look like.
-   Anyway, a complete definition is not required and some values could be missing
-   (AutoYaST should propose a reasonable default).
+2. Semi-automatic partitioning. The user would like to set some basic settings.
+   E.g., a user would like to use LVM but has no idea about how to configure
+   devices and partitions.
+3. Expert partitioning. The user specifies how the layout should look.
+   However, a complete definition is not required, and AutoYaST should propose
+   reasonable defaults for missing parts.
```
Comment (jreidinger) [inline doc/autoyast.md]: I am not sure, but is it similar to also installation where is 1. next, 2. guided proposal and 3.expert partitioner?

File: doc/autoyast.md (hunk 2 above)
Comment (imobachgs) [reply doc/autoyast.md]: Yes, I am not sure either. However, I prefer to point to the right place even if it is outdated. If that's the case, you can investigate how the code has changed and try to find the right place again.

File: doc/autoyast.md (hunk 3 above)
Comment (imobachgs) [reply doc/autoyast.md]: Yes, it is similar. The main difference is that in level 3, AutoYaST is able to figure out a lot of stuff by itself. For instance, this is enough:

```xml
<partitioning config:type="list">
  <drive>
    <device>/dev/sda</device>
    <use>all</use>
  </drive>
</partitioning>
```

**Pull request merged**